pkg/cnab/cnab-to-oci/registry.go

```diff
@@ -63,11 +63,11 @@ func (r *Registry) PushBundle(bun *bundle.Bundle, tag string, insecureRegistry b
 
 	resolver := r.createResolver(insecureRegistries)
 
-	err = remotes.FixupBundle(context.Background(), bun, ref, resolver, remotes.WithEventCallback(r.displayEvent))
+	rm, err := remotes.FixupBundle(context.Background(), bun, ref, resolver, remotes.WithEventCallback(r.displayEvent), remotes.WithAutoBundleUpdate())
```

Comment:
`cnab-to-oci` checks if the digest of any image from the bundle is different after pushed in the registry, and by default does _not_ mutate the bundle with the new image digests.

`remotes.WithAUtoBundleUpdate()` ensures the new digest is added to the bundle and pushed to the registry.

The main question here is whether there should be a flag to enforce the digest check and error out if any digest does not match.

Reply:
When @jeremyrickard and I have been discussing what happens during publish is that what is built and pushed is an immutable bundle by default (as best we can, scripts are gonna do what they do):

* The images referenced use digests, not tags
* Mixins have baked in their assets, e.g. the helm chart referenced is in the invocation image

https://github.com/deislabs/porter/issues/202

Given that, I vote that Porter always enforces digests. 

We can always make things more configurable later.  I don't want to encourage people to use a flag like that and  it will be confusing for most users. They'd have to dig into the docs to figure out what it even did, why they'd want to use it, etc. It's something that would be more suitable for duffle than Porter IMO.

Reply:
*yes gif*

Reply:
@radu-matei Do we have the right plumbing in place to add that check about enforcing digests in this PR? I would prefer to not merge the "fixup" behavior, and instead merge the PR with the "check and return error" behavior instead.